### PR TITLE
Multi commit operation base class

### PR DESCRIPTION
**Pull request by tidy-dev**
## Description

This class is an abstract class that any operation we add can extend so any operation can use this shared logic but can also have logic specific to that operation. 

Note: Dispatcher method for showing conflict dialog banner will be further updated in later PR's. More a less a partial place holder for this class.

## Release notes
Notes: no-notes

### CODE REVIEW

File: app/src/lib/stores/app-store.ts [hunk 1]
```diff
@@ -6376,25 +6376,6 @@ export class AppStore extends TypedBaseStore<IAppState> {
     targetBranch: Branch,
     commits: ReadonlyArray<Commit>
   ): void {
-    this.repositoryStateCache.updateMultiCommitOperationState(
```
Comment (tidy-dev) [inline app/src/lib/stores/app-store.ts:6379]: Have to refactor to use initializer method - done in a different PR; just removing for now.

**Review by sergiou87 — APPROVED** [review]
Looks good! Awesome job! 🙌
